src/js/game/hud/parts/building_placer.js

```diff
@@ -128,11 +128,17 @@ export class HUDBuildingPlacer extends BaseHUDPart {
         }
 
         // Deletion
-        if (button === enumMouseButton.right && !this.currentMetaBuilding.get()) {
+        if (
+            button === enumMouseButton.right &&
+            (!this.currentMetaBuilding.get() ||
+                !this.root.app.settings.getSetting("abortPlacementOnDeletion"))
```

Comment:
Could you change this into `getAllSettings().abortPlacementOnDeletion`? This will make sure the linter can validate it